### DISCUSSION TITLE
comparing input_data_version with upstream asset data_version

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#19282] [Q&A]
Hi,

I have an upstream asset (`up_asset`) that sets a user defined data_version in the `Output` on materialization, along with an explicit code_version='v1' . I can see both of these value in the UI after materializing `up_asset`. However, when a downstream asset `down_asset` materializes, the value of `input_data_version/up_asset` that is generated, is _not_ equal to the data_version` of `up_asset`.

In the code for `down_asset`, I want to determine if the up_stream has changed since the last materialization of `down_asset`.

How can I compare `input_data_version/up_asset` from the last materialization of `down_asset`, with the `dagster/data_version` and `dagster/code_version` from most recent materialization of `up_asset`.

**Note** both assets are partitioned daily

**Comment by the4thamigo-uk** [comment]
Safest option was to repurpose `CachingStaleStatusResolver` from the dagster internals

**Comment by DennisSchwartz** [comment]
I have the same issue, it felt quite surprising that the `input_data_version/<asset_key>` in the downstream asset materialization does not match the `data_version` tag of the upstream asset!

I will try to look into the `CachingStaleStatusResolver` now, hopefully that can provide a workaround for me as well.

**Comment by easontm** [comment]
Probably caused by #26563
